api/base/filters.py

```diff
@@ -20,53 +33,167 @@ def filter_queryset(self, request, queryset, view):
 
-class FilterMixin(object):
-    """ View mixin with helper functions for filtering. """
+    TIME_FIELDS = (ser.DateTimeField, ser.DateField)
+    DATETIME_PATTERN = re.compile(r'^\d{4}\-\d{2}\-\d{2}(?P<time>T\d{2}:\d{2}(:\d{2}(:\d{1,6})?)?)$')
 
-    DEFAULT_OPERATOR = 'eq'
+    COMPARABLE_FIELDS = NUMERIC_FIELDS + TIME_FIELDS
+
+    LIST_FIELDS = (ser.ListField, JSONAPIListField)
 
     def __init__(self, *args, **kwargs):
         super(FilterMixin, self).__init__(*args, **kwargs)
         if not self.serializer_class:
             raise NotImplementedError()
 
-    def is_filterable_field(self, key):
-        try:
-            return key.strip() in self.serializer_class.filterable_fields
-        except AttributeError:
-            return key.strip() in self.serializer_class._declared_fields
+    def _get_default_operator(self, field):
+        return self.DEFAULT_OPERATOR_OVERRIDES.get(type(field), self.DEFAULT_OPERATOR)
+
+    def _get_field_or_error(self, field_name):
+        """
+        Check that the attempted filter field is valid
+
+        :raises InvalidFilterError: If the filter field is not valid
+        """
+        if field_name not in self.serializer_class._declared_fields:
+            raise InvalidFilterError(detail="'{0}' is not a valid field for this endpoint.".format(field_name))
+        if field_name not in getattr(self.serializer_class, 'filterable_fields', {}):
+            raise InvalidFilterFieldError(attribute=field_name)
+        return self.serializer_class._declared_fields[field_name]
+
+    def _validate_operator(self, field, op):
+        """
+        Check that the operator and field combination is valid
+
+        :raises InvalidFilterComparisonType: If the query contains comparisons against non-date or non-numeric fields
+        :raises InvalidFilterMatchType: If the query contains comparisons against non-string or non-list fields
+        :raises InvalidFilterOperator: If the filter operator is not a member of self.COMPARISON_OPERATORS
+        """
+        if op not in set(self.MATCH_OPERATORS + self.COMPARISON_OPERATORS + (self.DEFAULT_OPERATOR, )):
+            raise InvalidFilterOperator(value=op)
+        if op in self.COMPARISON_OPERATORS:
+            if type(field) not in self.COMPARABLE_FIELDS:
+                raise InvalidFilterComparisonType(attribute=field.field_name)
+        if op in self.MATCH_OPERATORS:
+            if type(field) not in self.MATCHABLE_FIELDS:
+                raise InvalidFilterMatchType(attribute=field.field_name)
+
+    def _parse_date_param(self, field, field_name, op, value):
+        """
+        Allow for ambiguous date filters. This supports operations like findings Nodes created on a given day
+        even though Node.date_created is a specific datetime.
+
+        :return list<dict>: list of one (specific datetime) or more (date range) parsed query params
+        """
+        time_match = self.DATETIME_PATTERN.match(value)
```

Comment:
This is very edge case, but I was concerned with being unable to detect the difference between:
`[date_created]=2015-5-5T00:00:00`
and 
`[date_created]=2015-5-5`

The subtle difference being in both get parsed to the same `datetime`, but use different degrees os specificity.